### ISSUE TITLE
Very old Blockchain.com wallet The filesize is 328 bytes

### ISSUE THREAD

**Issue by AndrewEggg** [3rdIteration/btcrecover#216]
Starting btcrecover 1.11.0-Cryptoguide on Python 3.8.4 64-bit, 21-bit unicodes, 64-bit ints
Error: unrecognized wallet format; heuristic parser(s) reported:
    WalletPywallet: Unrecognized pywallet format (can't find mkey)
    WalletMultiBitHD: MultiBit HD wallet files must be named mbhd.wallet.aes
    WalletBlockchain: Encrypted Blockchain data length is not divisible by the encryption blocksize (16)

I got this error above when running the btcrecover please help.

**Comment by 3rdIteration** [comment]
What do you mean by "very old", if you open the file with a text editor, is there any recognizable structured text in there or just a bunch of characters?

**Comment by AndrewEggg** [comment]
The wallet is created in early 2012. There are two lines of encrypted text.  It seems I have same problem with this https://github.com/gurnec/btcrecover/issues/101 But instrction does not work I don't know how to do it.

**Comment by 3rdIteration** [comment]
If you run the original code from gurnec, do you get the entropy error or
the one that you posted about the length?

**Comment by AndrewEggg** [comment]
The error I posted above is when I run your btcrecover.py to find the password of the wallet.aes.json.
When I type the solution of gurnec I got a syntax error.

**Comment by AndrewEggg** [comment]
You mean I will use the btcrecover.py of gurnec?

**Comment by AndrewEggg** [comment]
It seems btcrecover does not recognize the format of the wallet.aes.json because it very old version.

**Comment by 3rdIteration** [comment]
Which exact error do you get when you use gurnec's version with your
wallet. I'm trying to determine whether you actually have the same issue as
us described in the link you provided.

**Comment by 3rdIteration** [comment]
Also, does it let you create a wallet extract if you try?

**Comment by AndrewEggg** [comment]
Gurnecs wallet extract does not work.  I got this error.
Traceback (most recent call last):
  File "download-blockchain-wallet.py", line 29, in <module>
    import sys, os.path, atexit, uuid, urllib2, json, time
ModuleNotFoundError: No module named 'urllib2'

**Comment by 3rdIteration** [comment]
His downloader will not work, but if you use the wallet.aes.json file that you downloaded with this repo with gurnec's repo, what error do you get?

**Comment by AndrewEggg** [comment]
Gurnec's wallet extract does not work anymore. I got an error.

**Comment by AndrewEggg** [comment]
I got this error when running Gurnec's code.
Traceback (most recent call last):
  File "btcrecover.py", line 30, in <module>
    from btcrecover import btcrpass
  File "/home/x/Downloads/btcrecover-mastergurnec/btcrecover/__init__.py", line 2, in <module>
    from .btcrpass import *
  File "/home/x/Downloads/btcrecover-mastergurnec/btcrecover/btcrpass.py", line 35, in <module>
    import sys, argparse, itertools, string, re, multiprocessing, signal, os, cPickle, gc, \
ModuleNotFoundError: No module named 'cPickle'

Yes the wallet wallet.aes.json was recently downloaded using your extract scripts.

**Comment by 3rdIteration** [comment]
Ah riteo, will be a pain to get Gurnec's version going.

Have you tried creating a wallet extract using your wallet.aes.json file and the extract script for blockchain wallets that is part of my repo?

**Comment by AndrewEggg** [comment]
The wallet.aes.json is extracted using your repo. Can you please update the btcrecover to support older blockchain.info wallets. If my bitcoins is recovered I will tip you some.

**Comment by 3rdIteration** [comment]
I'm not talking about downloading wallet.aes.json... I am asking you to
help me support wallets like this and to try using the
extract-blockchain-main-data.py
<https://github.com/3rdIteration/btcrecover/blob/master/extract-scripts/extract-blockchain-main-data.py>
or
extract-blockchain-second-hash.py
<https://github.com/3rdIteration/btcrecover/blob/master/extract-scripts/extract-blockchain-second-hash.py>
extract
scripts. (These are not the same as the script that downloads the wallet
file)

Simply put, the wallet type cannot be supported without more information
about exactly what works and what doesn't. (And even then, adding it
without some samples of wallet files may not be possible)

It would also be useful to know what sorts of errors you get when you try
to import the wallet via the the blockchain.com legacy wallet recovery tool
here: https://login.blockchain.com/wallet/import-wallet (Curious what the
javascript console spits out)

**Comment by AndrewEggg** [comment]
There was no error when I Input blockchain.com legacy wallet recovery tool
here: https://login.blockchain.com/wallet/import-wallet. It asked for password. 

I tried this tool. https://github.com/3rdIteration/btcrecover/blob/master/extract-scripts/extract-blockchain-main-data.py
The error is below.

python3 extract-blockchain-main-data.py wallet.aes.json
Traceback (most recent call last):
  File "extract-blockchain-main-data.py", line 83, in <module>
    raise ValueError("Encrypted Blockchain data length not divisible by encryption blocksize (16)")
ValueError: Encrypted Blockchain data length not divisible by encryption blocksize (16)

**Comment by AndrewEggg** [comment]
When using this script. https://github.com/3rdIteration/btcrecover/blob/master/extract-scripts/extract-blockchain-second-hash.py
The error is below.
Traceback (most recent call last):
  File "extract-blockchain-second-hash.py", line 140, in <module>
    raise ValueError("Encrypted Blockchain data length is not divisible by the encryption blocksize (16)")
ValueError: Encrypted Blockchain data length is not divisible by the encryption blocksize (16

**Comment by 3rdIteration** [comment]
Riteo. Can you try and just delete the following lines from ./btcrecover/btcrpass.py

https://github.com/3rdIteration/btcrecover/blob/4016a9c0a7433c3ac94d97da6747b4257d6f14f4/btcrecover/btcrpass.py#L1806-L1807

**Comment by AndrewEggg** [comment]
I got a new error after I deleted it.

Starting btcrecover 1.11.0-Cryptoguide on Python 3.8.10 64-bit, 21-bit unicodes, 64-bit ints
Error: unrecognized wallet format; heuristic parser(s) reported:
    WalletPywallet: Unrecognized pywallet format (can't find mkey)
    WalletMultiBitHD: MultiBit HD wallet files must be named mbhd.wallet.aes
    WalletBlockchain: Doesn't look random enough to be an encrypted Blockchain wallet (only 7.1 bits of entropy per byte)

**Comment by 3rdIteration** [comment]
Interesting, just try changing the min entropy to 7.0 

https://github.com/3rdIteration/btcrecover/blob/4016a9c0a7433c3ac94d97da6747b4257d6f14f4/btcrecover/btcrpass.py#L1817

**Comment by sigkill** [comment]
I was researching due to a friend with a similar issue unable to retrieve some of the very old blockchain.com / blockchain.info paperwallets due to both the complexity and various nonstandard wordlists they used - it looks like there are several different versions of wordlists, as well as different javascript code used on their site AND they allowed users to set or NOT set a passphrase for their exported wallet.aes.json private key wallets.

I found a few good links here: https://www.reddit.com/r/Bitcoin/comments/7kf60d/recovering_legacy_wallet_from_blockchaininfo_21/

and here:
https://keychainx.medium.com/how-to-recover-blockchain-info-15-17-19-21-mnemonic-seed-8691cb59425

With some minor info, and some of these links contained links for olther blockchain.info code - meaning some of it could be viewed and added to btcrecover's code.

Hopefully this is helpful, I am hunting to find the wordlists, if I do find them I'll post links and see if we can get a PR in place to add them to btcrecover if it's helpful. (Again these are blockchain.info specific, non-standard wordlists, thousands of sets of words.)

Hope this is helpful for anyone in similar circumstance.

**Comment by 3rdIteration** [comment]
Hey so @AndrewEggg did you manage to recover the wallet at all? (If so, it would be great to get a sample and improve BTCRecover's handling of these files)

**Comment by Drax4545** [comment]
I tried changing the entropy. the btcrecover worked.

**Comment by Drax4545** [comment]
Your suggestion worked for me.
